sentry-ruby/lib/sentry/open_telemetry/span_processor.rb

```diff
@@ -0,0 +1,63 @@
+module Sentry
+  module OpenTelemetry
+    class SpanProcessor
+      def initialize
+        @otel_span_map = {}
+      end
+
+      def on_start(otel_span, _parent_context)
+        return if from_sentry_sdk?(otel_span)
+
+        scope = Sentry.get_current_scope
+        parent_sentry_span = scope.get_span
+
+        sentry_span = if parent_sentry_span
+          parent_sentry_span.start_child(op: otel_span.name)
+        else
+          options = { name: otel_span.name, op: otel_span.name }
+          transaction = Sentry::Transaction.from_sentry_trace(scope.trace, **options) if scope.trace
+          scope.set_trace(nil)
+          Sentry.start_transaction(transaction: transaction, **options)
+        end
+
+        scope.set_span(sentry_span)
+        @otel_span_map[otel_span.context.span_id] = [sentry_span, parent_sentry_span]
+      end
+
+      def on_finish(otel_span)
+        current_scope = Sentry.get_current_scope
+        sentry_span, parent_span = @otel_span_map.delete(otel_span.context.span_id)
+        return unless sentry_span
+
+        sentry_span.set_op(otel_span.name)
+        current_scope.set_transaction_name(otel_span.name) if sentry_span.is_a?(Sentry::Transaction)
+
+        otel_span.attributes&.each do |key, value|
+          sentry_span.set_data(key, value)
+          if key == "db.statement"
+            sentry_span.set_description(value)
+          end
```

Comment:
We could apply more transformations here [like the collector does](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1fc794dfa65e55ab0faadeb16448f7c3daaf5c6f/exporter/sentryexporter/sentry_exporter.go#L226-L337). This was just an easy one to get us started.